sklearn/linear_model/quantile.py

```diff
@@ -0,0 +1,315 @@
+        be rewritten for every call to fit.
+        ``warm_start`` does not significantly speed up the convergence
+        if the model optimizes different cost functions, as ``gamma`` converges
+        to 0. It is therefore recommended to set small ``gamma``
+        if ``warm_start`` is set to True.
+
+    fit_intercept : bool, default True
+        Whether or not to fit the intercept. This can be set to False
+        if the data is already centered around the origin.
+
+    copy_X : boolean, optional, default True
+        If True, X will be copied; else, it may be overwritten.
+
+    normalize : boolean, optional, default False
+        This parameter is ignored when ``fit_intercept`` is set to False.
+        If True, the regressors X will be normalized before regression by
+        subtracting the mean and dividing by the l2-norm.
+
+    gamma : float, default 1e-2
+        Starting value for smooth approximation.
+        Absolute loss is replaced with quadratic for ``|error| < gamma``.
+        Lasso penalty is replaced with quadratic for ``|w| < gamma``.
+        ``gamma = 0`` gives exact non-smooth loss function.
+        The algorithm performs consecutive optimizations with gamma
+        decreasing by factor of ``gamma_decrease``,
+        until ``xtol`` criterion is met,
+        or until ``max_iter`` is exceeded.
+
+    gamma_decrease: float, default 0.1
+        The factor by which ``gamma`` is multiplied at each iteration.
+
+    n_gamma_decreases: int, default 10
+        Maximal number of iterations of approximation of the cost function.
+        At each iteration, ``gamma`` is multiplied by a factor
+        of ``gamma_decrease``
+
+    gtol : float, default 1e-4
+        The smooth optimizing iteration will stop when
+        ``max{|proj g_i | i = 1, ..., n}`` <= ``gtol``
+        where pg_i is the i-th component of the projected gradient.
+
+    xtol : float, default 1e-6
+        Global optimization will stop when ``|w_{t-1} - w_t|`` < ``xtol``
+        where w_t is result of t'th approximated optimization.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,)
+        Features got by optimizing the Huber loss.
+
+    intercept_ : float
+        Bias.
+
+    n_iter_ : int
+        Number of iterations that scipy.optimize.mimimize has run for.
+
+    References
+    ----------
+    .. [1] Koenker, R., & Bassett Jr, G. (1978). Regression quantiles.
+            Econometrica: journal of the Econometric Society, 33-50.
+
+    .. [2] Chen, C., & Wei, Y. (2005).
+           Computational issues for quantile regression.
+           Sankhya: The Indian Journal of Statistics, 399-417.
+    """
+
+    def __init__(self, quantile=0.5,
+                 max_iter=10000, alpha=0.0001, l1_ratio=0.0,
+                 warm_start=False, fit_intercept=True,
+                 normalize=False, copy_X=True,
+                 gamma=1e-2, gtol=1e-4, xtol=1e-6,
```

Comment:
How were the default values for `gtol`, `xtol` determined?

Reply:
I did manual experiments on a few generated datasets with normalized features, and found that these values suffice for the algorithm to converge. So yes, they are somehow arbitrary.

I can make some report describing how these parameters affect the results of computation. Do you think it is necessary?